frontend/src/components/AllQueues.js

```diff
@@ -87,16 +87,24 @@ function QueueTab(props) {
 const AllQueues = observer(() => {
   const queueStore = useContext(QueueStoreContext)
   const queueSize = Object.keys(queueStore.queues).length
-
-  useEffect(() => {
-    for (let i = 1; i <= queueSize; i++) queueStore.getQueue(i)
-  })
   const classes = useStyles()
   const [tabValue, setTabValue] = useState(0)
   function handleChange(event, newValue) {
     setTabValue(newValue)
   }
 
+  //Update all queues on first rendering
+  useEffect(() => {
+    for (let i = 1; i <= queueSize; i++) queueStore.getQueue(i)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
You don't need to disable eslint if you extract `getQueue` from `queueStore` (ie `const { getQueue } from queueStore` ) before all `useEffect`s and add `getQueue` to `useEffect` array. The component will become more efficient. `getQueue` shouldn't change but if it did, then `useMemo` or `useCallback` can be used.

Reply:
Nice.  I added queueStore.getQueue to the useEffect array.  It's just a function, so it should not change.  I got various mobx errors if I tried either using object destructuring as you suggested, or simply assigning const getQueue = queueStore.getQueue.  Oddly, the errors were different.  Is the efficiency gain just avoiding looking up the reference (or pointer or whatever you call it in this context) associated with queueStore.getQueue?

Reply:
Because the  getQueue assignment didn't work, the linter wanted me to include queueStore in the useEffect array.  I added it and took out the sub-dependencies, (i..e. getQueue) and it looks okay,  I didn't do this from the outset because I had a vague notion that it would degrade performance, but that's not based on evidence.

Reply:
What type of errors? Usually you would destructure properties and then use it in all proceeding functions, otherwise, it involves multiple lookups in the object for the same properties. I believe that's eslint and airbnb's recommended guidelines. Anyway, if it works, then it works.